File: schemas/schemasv1/terminal_record.go
package schemasv1

type TerminalRecordSchema struct {
	ResourceSchema
	Creator       *UserSchema         `json:"creator"`
	Organization  *OrganizationSchema `json:"organization"`
	Cluster       *ClusterSchema      `json:"cluster"`
	Deployment    *DeploymentSchema   `json:"deployment"`
	Resource      *ResourceSchema     `json:"resource"`
	PodName       string              `json:"pod_name"`
	ContainerName string              `json:"container_name"`
}

type TerminalRecordListSchema struct {
	BaseListSchema
	Items []*TerminalRecordSchema `json:"items"`
}

File: dashboard/src/schemas/terminal_record.ts
import { IDeploymentSchema } from '@/schemas/deployment'
import { IClusterSchema } from '@/schemas/cluster'
import { IOrganizationSchema } from '@/schemas/organization'
import { IUserSchema } from '@/schemas/user'
import { IResourceSchema } from '@/schemas/resource'

export interface ITerminalRecordSchema extends IResourceSchema {
    creator?: IUserSchema
    organization?: IOrganizationSchema
    cluster?: IClusterSchema
    deployment?: IDeploymentSchema
    resource?: IResourceSchema
    pod_name: string
    container_name: string
}
